Makefile

```diff
@@ -22,17 +37,98 @@ help:
 	@echo "                                      V E C T O R"
 	@echo ""
 	@echo "---------------------------------------------------------------------------------------"
+	@echo -e "Nix user? You can use \033[0;33m\`direnv allow .\`\033[0m or \033[0;33m\`nix-shell --pure\`\033[0m"
+	@echo -e "Want to use \033[0;33m\`docker\`\033[0m or \033[0;33m\`podman\`\033[0m? See \033[0;33m\`ENVIRONMENT=true\`\033[0m commands. (Default \033[0;33m\`CONTAINER_TOOL=docker\`\033[0m)"
 	@echo ""
 	@awk 'BEGIN {FS = ":.*##"; printf "Usage: make \033[36m<target>\033[0m\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-46s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 
-##@ Default
+##@ Environment
+# We use a volume here as non-Linux hosts are extremely slow to share disks, and Linux hosts tend to get permissions clobbered.
+define ENVIRONMENT_EXEC
```

Comment:
I feel like these (`ENVIRONMENT_EXEC`, `ENVIRONMENT_COPY_ARTIFACTS` and `ENVIRONMENT_PREPARE`) are good candidates to be manually invokable commands at `scripts/`. But then, it's very like I personally wouldn't be invoking them. They'll be useful for automation that you can't drive from the environment shell but that you want to run in the environment - for instance, editor integration and etc...

Reply:
I generally suggest using Nix, all `ENVIRONMENT_*` things are only really existing to support the choice to using Docker for a dev env, which is not a good choice.

Reply:
Let's make `environment_exec` a script!

Reply:
I agree :smiley: Yeah, we probably don't want to add more flexibility into docker since we have nix. :+1: